File: v6/sprefa-extract/tests/fixtures/go_findings/closure_mirror/closure_mirror.go
// Expected: every closure-caller edge (a call whose covering def is a Lambda)
// gets ONE mirror edge onto the innermost NAMED enclosing def; nested closures
// mirror to `outer`, never to an outer closure; the package-level literal's
// body mints no def, so its call emits no row at all.
// Observed: the three closure-caller sites (helper/wrap in step, helper in
// inner, helper at package level) — the two in-named-fn closures each carry a
// mirror to `outer`; the package-level site resolves to nothing.
package mirror

var pkgLevel = func() {
	helper()
}

func helper() {}

func wrap() {}

func outer() {
	step := func() {
		helper()
		wrap()
	}
	inner := func() {
		step()
		helper()
	}
	step()
	inner()
}

File: v6/sprefa-extract/tests/fixtures/ts5_findings/closure_mirror/closure_mirror.ts
// Expected: every closure-caller edge (a call whose covering def is a Lambda)
// gets ONE mirror edge onto the innermost NAMED enclosing def, the walk
// `oracle_ts.mjs` `enclosingName` does; nested anonymous arrows mirror to
// `outer`, never to an outer arrow, and a module-level arrow mirrors to
// `<module>`.
// Observed at HEAD: the closure rows are the only rows; no mirror exists.

export function helper(x: number): number {
  return x;
}

export function wrap(): number {
  return 1;
}

export function run(fn: () => void): void {
  fn();
}

export function outer(): void {
  run(() => {
    helper(1);
    run(() => {
      wrap();
    });
  });
}

run(() => {
  helper(2);
});
